File: frontend/types/history.ts
export interface HistoryItem {
  id: string;
  user_id: string;
  project_id: string | null;
  video_title: string;
  description: string | null;
  tags: string[] | null;
  youtube_video_id: string;
  youtube_url: string;
  privacy_status: "private" | "unlisted" | "public";
  published_at: string;
  status: string;
  thumbnail_url: string | null;
  content_type: string;
  created_at: string;
}

export interface HistoryListResponse {
  items: HistoryItem[];
  total: number;
  page: number;
  page_size: number;
  has_next: boolean;
}

File: backend/app/schemas/history.py
from __future__ import annotations

import uuid
from datetime import datetime
from typing import List

from pydantic import BaseModel, ConfigDict


class HistoryItemResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    project_id: str | None
    video_title: str
    description: str | None
    tags: List[str] | None
    youtube_video_id: str
    youtube_url: str
    privacy_status: str
    published_at: datetime
    status: str
    thumbnail_url: str | None
    content_type: str = "video"
    created_at: datetime


class HistoryListResponse(BaseModel):
    items: List[HistoryItemResponse]
    total: int
    page: int
    page_size: int
    has_next: bool
